build-tools/packages/build-cli/src/commands/generate/manifest.ts

```diff
@@ -0,0 +1,226 @@
+		const flags = this.flags;
+
+		// Authorization header for Azure DevOps
+		const authHeader = `Basic ${Buffer.from(`:${flags.ado_pat}`).toString("base64")}`;
+
+		if (authHeader === undefined || authHeader === null) {
+			this.error("Check your ADO Personal Access Token. It maybe incorrect or expired.");
+		}
+
+		try {
+			const buildNumber = await getFirstSuccessfulBuild(
+				authHeader,
+				flags.organization,
+				flags.project,
+				flags.sourceBranch,
+				this.logger,
+			);
+			if (buildNumber !== undefined) {
+				this.log(
+					`Most successful build number for the last 24 hours for ${flags.sourceBranch} branch: ${buildNumber}`,
+				);
+				const devVersion = await fetchDevVersionNumber(
+					authHeader,
+					flags.organization,
+					flags.project,
+					buildNumber,
+					this.logger,
+				);
+				if (devVersion !== undefined) {
+					this.log(`Fetched dev version: ${devVersion}`);
+					await generateManifestFileForDevReleases(devVersion, this.logger);
+				}
+			} else if (buildNumber === undefined) {
+				this.log(
+					`No successful build found for ${flags.sourceBranch} branch in the last 24 hours`,
+				);
+			}
+		} catch (error: unknown) {
+			this.errorLog(`Error creating manifest file: ${error}`);
+		}
+	}
+}
+
+/**
+ * Fetches the first successful build number in the last 24 hours.
+ * @returns The build number if successful, otherwise undefined.
+ */
+async function getFirstSuccessfulBuild(
+	authHeader: string,
+	organization: string,
+	project: string,
+	sourceBranch: string,
+	log?: Logger,
+): Promise<string | undefined> {
+	const ADO_BASE_URL = `https://dev.azure.com/${organization}/${project}/_apis/build/builds?api-version=7.0`;
+	try {
+		const response = await fetch(ADO_BASE_URL, { headers: { Authorization: authHeader } });
+		const data = await response.json();
+
+		const twentyFourHoursAgo = new Date();
+		twentyFourHoursAgo.setHours(twentyFourHoursAgo.getHours() - 24);
+
+		const successfulBuilds = data.value.filter(
+			(build: IBuildDetails) =>
+				build.definition.name === "Build - client packages" &&
+				build.status === "completed" &&
+				build.result === "succeeded" &&
+				build.sourceBranch === `refs/heads/${sourceBranch}` &&
+				new Date(build.finishTime) >= twentyFourHoursAgo,
+		);
+
+		return successfulBuilds.length > 0
+			? (successfulBuilds[0].buildNumber as string)
+			: undefined;
+	} catch (error) {
+		log?.errorLog("Error fetching successful builds:", error);
+		return undefined;
+	}
+}
+
+/**
+ * Fetches the dev version number released in the build feed.
+ * @param buildNumber - The build number.
+ * @returns The dev version number if found, otherwise undefined.
+ */
+async function fetchDevVersionNumber(
+	authHeader: string,
+	organization: string,
+	project: string,
+	buildNumber: string,
+	log?: Logger,
+): Promise<string | undefined> {
+	const REGISTRY_URL = `https://pkgs.dev.azure.com/${organization}/${project}/_packaging/build/npm/registry/`;
```

Comment:
I would make this function take a registry URL instead of building it. That way you can test it against any URL.